packages/x-data-grid/src/components/virtualization/GridTopContainer.tsx

```diff
@@ -29,13 +35,17 @@ const Element = styled('div')({
 });
 
 export function GridTopContainer(props: React.HTMLAttributes<HTMLDivElement>) {
+  const { children, ...otherProps } = props;
   const classes = useUtilityClasses();
 
   return (
     <Element
-      {...props}
-      className={clsx(classes.root, props.className, gridClasses['container--top'])}
+      {...otherProps}
+      className={clsx(classes.root, otherProps.className, gridClasses['container--top'])}
       role="presentation"
-    />
+    >
+      <GridPanelAnchor role="presentation" data-id="gridPanelAnchor" />
```

Comment:
Thanks @oukunan, nice catch!

Unfortunately, putting `GridPanelAnchor` here causes an issue when scrolling horizontally:
https://codesandbox.io/p/sandbox/silly-firefly-hmtdfn


https://github.com/mui/mui-x/assets/13808724/15e2773b-d152-44e7-bce0-9f6e1a819de7

I'll update the PR with a fix.